### DISCUSSION TITLE
Add a way to collapse sidebar on desktop screen

### DISCUSSION THREAD

**Discussion by iapparatus** [livewire/flux#877] [Feature Requests] [answered]
It would be useful to have a button to close/collapse the sidebar on a desktop-size screen when it is open by default.

**Comment by igmltd** [comment]
Any luck with this? Any help would be much appreciated  - thanks

**Comment by aymanatmeh** [comment]
It's helpful to have this feature, especially when you have both a side menu and a navigation menu. It allows for collapsing the side menu, enhancing usability.

<img width="1509" alt="Screenshot 2025-01-26 at 10 48 27 am" src="https://github.com/user-attachments/assets/c93dcc95-64c2-4ca8-a73b-fb4abcf4df41" />

**Reply by joffreypersia** [reply]
This is not a good UX design as it shift the main content to the right and shrink it eventually

**Comment by igmltd** [comment]
Shadcn have done a great job here

https://ui.shadcn.com/docs/components/sidebar

**Comment by 0hra** [comment]
Would be nice with a collapsed/stashed by default option as well.

**Comment by igmltd** [comment]
Yeah, that would also make it function like the new shadcn starter kit announced at Laracon EU

**Comment by brnhache** [comment]
Im just forcing it to hide with some javascript. kinda crazy how its not a thing on desktop - I made separate views for mobile and desktop just to handle this.

Sidebar with close button
`<flux:sidebar class="min-h-screen shadow-md bg-zinc-200 dark:border-zinc-700 dark:bg-zinc-900"
    id="sidebar" sticky x-cloak
    x-show="!sessionStorage.getItem('sidebarState') || sessionStorage.getItem('sidebarState') === 'open'">
    <flux:icon.x-mark
        class="hidden h-10 w-10 cursor-pointer rounded-sm p-1 hover:border hover:border-slate-300 hover:shadow-md lg:block"
        id="close-sidebar-lg" inset="left" />`
        
Top Navbar with open button
`<flux:header class="!justify-between shadow-md bg-zinc-200 dark:border-zinc-700 dark:bg-zinc-900"
    id="top-nav" sticky>
    <div>
        <flux:icon.bars-3
        class="hidden h-10 w-10 cursor-pointer rounded-sm p-1 hover:border hover:border-slate-300 hover:shadow-md lg:block"
        id="open-sidebar-lg" inset="left" x-cloak
        x-show="sessionStorage.getItem('sidebarState') === 'closed'" />
    </div>`
    
Jquery to toggle     
`<script>
    $('#open-sidebar-lg').click(() => {
        $('#sidebar').show();
        $('#open-sidebar-lg').hide();
        sessionStorage.setItem('sidebarState', 'open');
    });
    $("#close-sidebar-lg").click(() => {
        $('#sidebar').hide();
        $('#open-sidebar-lg').show();
        sessionStorage.setItem('sidebarState', 'closed');
    });
    </script>`

**Comment by 0hra** [comment]
@joshhanley Is this feature being considered?

**Reply by joshhanley** [reply]
@0hra at the moment, no, as we're working on bug fixes since the V2 release.

**Comment by benjibee** [comment]
![image](https://github.com/user-attachments/assets/724754c8-25cc-40f0-a08e-0358ad8c578e)

Thought [this tweet](https://x.com/dnak0v/status/1902464826602844378) might be relevant here.

**Comment by snapey** [comment]
I would like the option to collapse the sidebar to just its icons, with open and close icons in the trigger

**Comment by elliottcoe** [comment]
Adding to this chat, I think having the ability to toggle the sidebar on desktop is important.

**Comment by stuartbrameld** [comment]
To add to this, I'm increasingly finding it useful to have a right-hand (AI) chat panel for various tasks, but having this in addition to a full left-hand sidebar results in a lot of lost screen real estate hence the option to stash the nav bar would be nice.

**Comment by hellodit** [comment]
is anyone know the solution?

**Comment by FranciscoDumont** [comment]
Hey everyone! 👋🏻

I found a workaround to make the sidebar toggleable on desktop screens. Here’s exactly what I did:

1. Ran the command:
    ```bash
    php artisan flux:publish
    ```
    and selected the option to publish the `Layout`.

2. In the published layout file `resources/views/flux/sidebar/index.blade.php`, I made the following changes:
    - Replaced both instances of `1024` (lines 21 and 35) with a high value like `999999` to treat all screens as "desktop" and avoid triggering mobile-specific behavior.
    - On line 24: changed `max-lg` to `max-7xl`
    - On line 25: changed `lg:translate` to `7xl:translate`

3. Searched for `<flux:sidebar.toggle>` components in my codebase and removed the `lg:hidden` class wherever it was used


It’s not a perfect solution, but it worked well enough for my needs. Hope this helps someone else trying to do the same!

**Comment by aigorlaxy** [comment]
I would like it too!

**Comment by bhorlings-pcg** [comment]
This is a need!

**Comment by Uhasith** [comment]


**Reply by joffreypersia** [reply]
lol
Caleb presented yesterday Livewire 4 that will improve Flux in many ways.
Trust the process and a collapsible sidebar is just an alpine work.

**Reply by Uhasith** [reply]
> lol Caleb presented yesterday Livewire 4 that will improve Flux in many ways. Trust the process and a collapsible sidebar is just an alpine work.

61 people voting for this mate. At least guide us to temporary fix ? @joffreypersia

**Reply by joffreypersia** [reply]
Livewire is maintained solely by @joshhanley and @calebporzio (apart from the free, open-source portion). Since Josh has already responded in this thread, we can assume he is aware of this feature request. At the moment, it does not seem to be a priority, as both Caleb and Josh are currently focused on Livewire (which will impact Flux) and Blaze (which will also impact Flux from a broader perspective). For more information, check out the LaraconUS Day 2 stream on YouTube.

I do get a bit frustrated when people repeatedly request "Alpine wrapper components," especially when comments like "and it costs me $300" are made. These requests aren't directly related to Flux. It seems some don’t recognize the significant amount of work and thought required to develop actual Flux components, or where improvements and new components can truly benefit the product.

**Reply by joshhanley** [reply]
@Uhasith while I appreciate it may seem like a small thing, we've been non-stop working on and improving Flux since it's launch. We can't get to every feature and I can't guarantee a timeline on when features will be built. But rest assured we will get to it. We're committed to making Flux better, as demonstrated at Laracon US last week and are tackling what we feel is most important first. I do want to clarify one thing though - a feature request, like this one, is no guarantee it will get added as it's just a community suggestion. @hellodit this applies to your comment as well, we're fully aware of the feature requests we get and it hasn't slipped through, we've just prioritised other aspects of Flux's development. If either of you don't feel that is good enough, please email us at support@fluxui.dev and we can issue you a refund. Otherwise, Flux is what it is for now and it will continue to grow and be improved in time. Happy Fluxing! 🙂

**Reply by joshhanley** [reply]
Thanks for your support @joffreypersia! 😊 The comment you make is true, Flux is never going to be able to cover every scenario of what people are going to need. As such they need to get comfortable building parts that they need for those times that Flux hasn't got it covered yet.

**Comment by joffreypersia** [comment]
Guys, @joshhanley is cooking 
https://github.com/livewire/flux/compare/main...josh/sidebars
Just saying 😎

**Reply by bridgeyuwa** [reply]
> Guys, @joshhanley is cooking
> [main...josh/sidebars](https://github.com/livewire/flux/compare/main...josh/sidebars)
> Just saying 😎

Eagerly waiting for the meal. 😎

**Reply by Uhasith** [reply]
> Guys, @joshhanley is cooking [main...josh/sidebars](https://github.com/livewire/flux/compare/main...josh/sidebars) Just saying 😎

Yoooo finally 🥹

**Reply by joshhanley** [reply]
👀

**Comment by francisgradient** [comment]
I tried something to match the shadcn type of collapsible sidebar using alpinejs in the livewire starter kit and it doesn't mess the mobile ui. Tried to add animation for smoothness but it conflicts with the elements. 

**sidebar.blade.php**
```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}" class="dark">
    <head>
        @include('partials.head')
    </head>
    <body class="min-h-screen bg-white dark:bg-zinc-800" x-data="{ sidebarHidden: false }">
        <flux:sidebar sticky stashable x-bind:class="sidebarHidden ? 'border-e border-zinc-200 bg-zinc-50 dark:border-zinc-700 dark:bg-zinc-900 hidden' : 'border-e border-zinc-200 bg-zinc-50 dark:border-zinc-700 dark:bg-zinc-900 '">
            <flux:sidebar.toggle class="lg:hidden" icon="x-mark" />

            <a href="{{ route('dashboard') }}" class="me-5 flex items-center space-x-2 rtl:space-x-reverse" wire:navigate>
                <x-app-logo />
            </a>

            <flux:navlist variant="outline">
                <flux:navlist.group :heading="__('Platform')" class="grid">
                    <flux:navlist.item icon="home" :href="route('dashboard')" :current="request()->routeIs('dashboard')" wire:navigate>{{ __('Dashboard') }}</flux:navlist.item>
                </flux:navlist.group>
            </flux:navlist>

            <flux:spacer />

            <flux:navlist variant="outline">
                <flux:navlist.item icon="folder-git-2" href="https://github.com/laravel/livewire-starter-kit" target="_blank">
                {{ __('Repository') }}
                </flux:navlist.item>

                <flux:navlist.item icon="book-open-text" href="https://laravel.com/docs/starter-kits#livewire" target="_blank">
                {{ __('Documentation') }}
                </flux:navlist.item>
            </flux:navlist>

            <!-- Desktop User Menu -->
            <flux:dropdown class="hidden lg:block" position="bottom" align="start">
                <flux:profile
                    :name="auth()->user()->name"
                    :initials="auth()->user()->initials()"
                    icon:trailing="chevrons-up-down"
                />

                <flux:menu class="w-[220px]">
                    <flux:menu.radio.group>
                        <div class="p-0 text-sm font-normal">
                            <div class="flex items-center gap-2 px-1 py-1.5 text-start text-sm">
                                <span class="relative flex h-8 w-8 shrink-0 overflow-hidden rounded-lg">
                                    <span
                                        class="flex h-full w-full items-center justify-center rounded-lg bg-neutral-200 text-black dark:bg-neutral-700 dark:text-white"
                                    >
                                        {{ auth()->user()->initials() }}
                                    </span>
                                </span>

                                <div class="grid flex-1 text-start text-sm leading-tight">
                                    <span class="truncate font-semibold">{{ auth()->user()->name }}</span>
                                    <span class="truncate text-xs">{{ auth()->user()->email }}</span>
                                </div>
                            </div>
                        </div>
                    </flux:menu.radio.group>

                    <flux:menu.separator />

                    <flux:menu.radio.group>
                        <flux:menu.item :href="route('settings.profile')" icon="cog" wire:navigate>{{ __('Settings') }}</flux:menu.item>
                    </flux:menu.radio.group>

                    <flux:menu.separator />

                    <form method="POST" action="{{ route('logout') }}" class="w-full">
                        @csrf
                        <flux:menu.item as="button" type="submit" icon="arrow-right-start-on-rectangle" class="w-full">
                            {{ __('Log Out') }}
                        </flux:menu.item>
                    </form>
                </flux:menu>
            </flux:dropdown>
        </flux:sidebar>
        <flux:header sticky class=" shadow-md max-lg:hidden bg-zinc-900">
            <div class="cursor-pointer group hover:bg-zinc-700 rounded-lg p-2" x-on:click="sidebarHidden = !sidebarHidden">
                <svg class="w-4 h-4 stroke-gray-400 group-hover:stroke-white" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" fill="none" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
                    <rect x="3" y="3" width="18" height="18" rx="2" ry="2" />
                    <line x1="9" y1="3" x2="9" y2="21" />
                </svg>
            </div>
        </flux:header>

        <!-- Mobile User Menu -->
        <flux:header class="lg:hidden">
            <flux:sidebar.toggle class="lg:hidden" icon="bars-2" inset="left" />

            <flux:spacer />

            <flux:dropdown position="top" align="end">
                <flux:profile
                    :initials="auth()->user()->initials()"
                    icon-trailing="chevron-down"
                />

                <flux:menu>
                    <flux:menu.radio.group>
                        <div class="p-0 text-sm font-normal">
                            <div class="flex items-center gap-2 px-1 py-1.5 text-start text-sm">
                                <span class="relative flex h-8 w-8 shrink-0 overflow-hidden rounded-lg">
                                    <span
                                        class="flex h-full w-full items-center justify-center rounded-lg bg-neutral-200 text-black dark:bg-neutral-700 dark:text-white"
                                    >
                                        {{ auth()->user()->initials() }}
                                    </span>
                                </span>

                                <div class="grid flex-1 text-start text-sm leading-tight">
                                    <span class="truncate font-semibold">{{ auth()->user()->name }}</span>
                                    <span class="truncate text-xs">{{ auth()->user()->email }}</span>
                                </div>
                            </div>
                        </div>
                    </flux:menu.radio.group>

                    <flux:menu.separator />

                    <flux:menu.radio.group>
                        <flux:menu.item :href="route('settings.profile')" icon="cog" wire:navigate>{{ __('Settings') }}</flux:menu.item>
                    </flux:menu.radio.group>

                    <flux:menu.separator />

                    <form method="POST" action="{{ route('logout') }}" class="w-full">
                        @csrf
                        <flux:menu.item as="button" type="submit" icon="arrow-right-start-on-rectangle" class="w-full">
                            {{ __('Log Out') }}
                        </flux:menu.item>
                    </form>
                </flux:menu>
            </flux:dropdown>
        </flux:header>

        {{ $slot }}

        @fluxScripts
    </body>
</html>

```


https://github.com/user-attachments/assets/7f2f26ce-7724-4056-83ab-b55e424b2147

**Comment by Uhasith** [comment]
Finally ❤️

[Doc Link](https://fluxui.dev/layouts/sidebar#collapsible-sidebar)

**Comment by joshhanley** [comment, marked as answer]
We've just released collapsible sidebars in v2.3.0!

See the blog post here https://fluxui.dev/blog/2025-09-03-collapsible-sidebars
And you can find the documentation here https://fluxui.dev/layouts/sidebar

**Reply by lcamero** [reply]
This looks cool!

I've been trying to get it to work but for some reason the toggle won't do anything when clicked, apparently there is no click event handler when I look at it in devtools.

Must be something in my setup though, because I copied and pasted the code from the docs (which clearly works) and it just won't click. 

I tried clearing the view cache in laravel but still nothing, got any idea what else I could check?

I am using the livewire-starter-kit but I don't see why that'd be relevant for this necessarily

<img width="600" height="762" alt="image" src="https://github.com/user-attachments/assets/7b77a067-23e0-48b4-8128-a6c64e1d4462" />

**Reply by joshhanley** [reply]
@lcamero the click listener will be on the button. Please open an issue if you are still having problems. Thanks!
